### DISCUSSION TITLE
Set "custom" timeout mid-run

### DISCUSSION THREAD

**Discussion by kskarra** [getgauge/gauge#2054] [General]
Trying to run a while-loop in a test to wait for certain processes to finish before going ahead. Is there a clever way to set the timeout for test/scenario while this loop is running? 
As I've understood this is depending on the js.properties timeout, but can't really find anything regarding this in the documentation.

**Comment by sswaroopgupta** [comment]
JS timeout is the timeout for a step. Irrespective of whether there is a waitFor within the step or not, Taiko gives JSTimeout milliseconds for the step to execute. So if the code in the step takes more than the configured time then it throws a timeout exception.

If you want to wait for a given condition, you should explore waitForNavigation or waitForEvents. Here is the documentation https://docs.taiko.dev/api/tap/. Hope this helps

**Reply by kskarra** [reply]
Thanks for the reply!
It's what I figured as well, but whichever Taiko waitFor function being used will time out when the JS Timeout is reached.
And I assume having the JS timeout set to 10-15 minutes isn't ideal? Its been like 45 sec as long as we've run it.

**Reply by sswaroopgupta** [reply]
How long does the page take to complete the network calls when you execute manually should give you an idea of what you will need to set the JSTimeout for a given env. The default config is at env\default\js.properties. 

PS:- If waitFor events are not working for you, you can wait for an overlay to disappear.
For example:- `await waitFor(async () => !(await $("overlay").exists()))` and then perform the next action. Hope this helps
